libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackDetailsFragment.kt

```diff
@@ -0,0 +1,498 @@
+                FeedbackEvent.ROUTE_HAD_ROADS_TOO_NARROW_TO_PASS,
+                R.string.mapbox_feedback_description_route_had_roads_too_narrow_to_pass
+            )
+        )
+        return list
+    }
+
+    private fun subTypeOfNotAllowed(): List<FeedbackSubTypeItem> {
+        val list: MutableList<FeedbackSubTypeItem> = ArrayList()
+        list.add(
+            FeedbackSubTypeItem(
+                FeedbackEvent.ROUTED_DOWN_A_ONE_WAY,
+                R.string.mapbox_feedback_description_routed_down_a_one_way
+            )
+        )
+        list.add(
+            FeedbackSubTypeItem(
+                FeedbackEvent.TURN_WAS_NOT_ALLOWED,
+                R.string.mapbox_feedback_description_turn_was_not_allowed
+            )
+        )
+        list.add(
+            FeedbackSubTypeItem(
+                FeedbackEvent.CARS_NOT_ALLOWED_ON_STREET,
+                R.string.mapbox_feedback_description_cars_not_allowed_on_street
+            )
+        )
+        list.add(
+            FeedbackSubTypeItem(
+                FeedbackEvent.TURN_AT_INTERSECTION_WAS_UNPROTECTED,
+                R.string.mapbox_feedback_description_turn_at_intersection_was_unprotected
+            )
+        )
+        return list
+    }
+
+    private fun subTypeOfRoadClosed(): List<FeedbackSubTypeItem> {
+        val list: MutableList<FeedbackSubTypeItem> = ArrayList()
+        list.add(
+            FeedbackSubTypeItem(
+                FeedbackEvent.STREET_PERMANENTLY_BLOCKED_OFF,
+                R.string.mapbox_feedback_description_street_permanently_blocked_off
+            )
+        )
+        list.add(
+            FeedbackSubTypeItem(
+                FeedbackEvent.ROAD_IS_MISSING_FROM_MAP,
+                R.string.mapbox_feedback_description_road_is_missing_from_map
+            )
+        )
+        return list
+    }
+
+    private val bottomSheetCallBack = object : BottomSheetCallback() {
+        override fun onSlide(bottomSheet: View, slideOffset: Float) {
+            hideSoftInput(bottomSheet)
+        }
+
+        override fun onStateChanged(bottomSheet: View, newState: Int) {
+        }
+    }
+
+    companion object {
+        /**
+         * Create a new instance of [FeedbackDetailsFragment] with given feedback items.
+         *
+         * @param feedbackItemList is a list of [CachedNavigationFeedbackEvent] which is generated by the user along with the route.
+         *  In feedback detail mode, user can provide more details about a feedback. Like providing feedback sub types and feedback description.
+         * @param feedbackFlowListener a listener will be called when the user is done with providing feedback details.
+         *  This listener will be passed to [FeedbackArrivalFragment] as the callback when the user finishes the arrival feedback.
+         * @param enableArrivalExperienceFeedback true to display a [FeedbackArrivalFragment] when the [FeedbackDetailsFragment] flow is done.
+         */
+        fun newInstance(
```

Comment:
```suggestion
        @JvmStatic
        fun newInstance(
```

That'd remove the need of `Companion` at the call-site.